src/framework/components/layout-group/layout-calculator.js

```diff
@@ -590,12 +590,14 @@ pc.extend(pc, function () {
         function getTraversalOrder(items, orderBy, descending) {
             items.forEach(assignIndex);
 
+            /* eslint-disable dot-location */
```

Comment:
I sense you're trying to tell me I've configured the dot-location rule incorrectly. 😉 TBH, I don't mind whether it's at the start of a line or at the end of the previous. The default for the rule is at the end of the line after the object so maybe that's why I went with that. Happy to change it globally. But I'd rather avoid disabling rules like this.

Reply:
Oh, haha, really sorry – I wasn't actually, I just misinterpreted the rule. I thought it was asking me to place all of the chained calls on the same line, i.e.

```
 return items.slice().sort(sortFn).map(getIndex);
```

And I felt that separating them was actually more readable in this case.

I've never actually considered the idea of placing dots on the same line as the object and then the next function call on the next line, as the rule actually dictates...it seems quite alien to me. The MDN documentation for [Promise](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/then), for example, places all of the dots on the same line as the function call whenever calls are chained:

```
Promise.reject()
  .then( () => 99, () => 42 )
  .then( solution => console.log( 'Resolved with ' + solution ) );
```

In any case I don't care as long as it's readable – will update.

Reply:
Hmm, doing a bit more research – the AirBnB and Google styleguides both recommend dot on next line:

- https://github.com/airbnb/javascript#whitespace--chains
- https://google.github.io/styleguide/jsguide.html#formatting-where-to-break

Perhaps it is more of a defacto standard. Might make sense to switch globally if it's not a massive pain?